### PR TITLE
measures processing time of each kind of gossip packets

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Gossip packets processing breakdown

#### Summary of Changes
Added metrics for measuring processing time of each kind of gossip packets.

### CODE REVIEW

File: core/src/cluster_info.rs [hunk 1]
```diff
@@ -196,6 +196,16 @@ impl Counter {
     fn clear(&self) -> u64 {
         self.0.swap(0, Ordering::Relaxed)
     }
+    fn measure_task<F, R>(&self, task: F) -> R
+    where
+        F: FnOnce() -> R,
+    {
+        let now = Instant::now();
```
Comment (carllin) [inline core/src/cluster_info.rs:203]: nit: We've been shifting to using `Measure`  and `add_measure` like here: https://github.com/solana-labs/solana/blob/master/core/src/cluster_info.rs#L1162-L1177 over `Instant`

**Review by carllin — APPROVED** [review]

**Review by sakridge — APPROVED** [review]
2nd preference for Measure, but lgtm

File: core/src/cluster_info.rs (hunk 1 above)
Comment (behzadnouri) [reply core/src/cluster_info.rs]: yes, that is what I am using elsewhere, but here, some of these functions have multiple points of `return` so adding a `self.stats.abc.add_measure(&mut timer)` at each of those `return` points is a bit verbose and clumsy.
I have updated this pull-request with a new `ScopedTimer` to automatically track the execution time of the function with just one line addition.

**Review by carllin — APPROVED** [review]

File: core/src/cluster_info.rs (hunk 1 above)
Comment (carllin) [reply core/src/cluster_info.rs]: cool thanks!

**Pull request merged**